### ISSUE TITLE
Using AddAuthenticationJwtBearer(signingOptions) with a base 64 symmetric signing key in ASP.NET 9

### ISSUE THREAD

**Issue by queryquandary** [FastEndpoints/FastEndpoints#844]
### Description
Getting an HTTP 401 response when using FastEndpoints.Security.AuthExtensions.AddAuthenticationJwtBearer() signing options in ASP.NET 9 to set up a symmetric signing key.

#### Notes:
- This only applies to **_ASP.NET Core 9_** and is not an issue in .NET8.
- I am working with local tokens using ```dotnet user-jwts``` tool.

### Steps to recreate:

1. In Program.cs, use AddAuthenticationJwtBearer() to set up JWT authentication with a symetric base 64 encoded signing key

``` c#
builder.Services
    .AddAuthenticationJwtBearer(signingOptions =>
    {
        signingOptions.SigningStyle = TokenSigningStyle.Symmetric;
        signingOptions.SigningKey = builder.Configuration["Auth:JwtSecret"]; // Base 64 encoded signing key
    })
    .AddAuthorization();

```

2. Make a request that requires auth
3. Get an HTTP 401 status

### Cause:
- FastEndpoints.Sercurity.AuthExtensions.AddAuthenticationJwtBearer is using Encoding.ASCII.GetBytes to convert a base 64 encoded string into a byte array.
https://github.com/FastEndpoints/FastEndpoints/blob/0f1f9df14301c455c1ac87c64f8cff8b56bc83a9/Src/Security/AuthExtensions.cs#L44

### Proposed repair in FastEndpoints.Security.AuthExtensions.cs:
- Use ```Convert.ToBase64String()``` instead of ```Encoding.ASCII.GetBytes()``` when creating the SymmetricSecurityKey
- This works for both .NET 8 and .NET 9

``` c#
key = new SymmetricSecurityKey(Convert.FromBase64String(sOpts.SigningKey));
```

### Workaround:
- Use bearerOptions in AddAuthenticationJwtBearer() to manually create your own SymmetricSecurityKey:
``` c#
builder.Services
    .AddAuthenticationJwtBearer(signingOptions =>
    {
        signingOptions.SigningStyle = TokenSigningStyle.Symmetric;
    },
    bearerOptions =>
    {
        var base64SigningKey = builder.Configuration["Auth:JwtSecret"];
        var key = new SymmetricSecurityKey(Convert.FromBase64String(base64SigningKey!));
        bearerOptions.TokenValidationParameters.IssuerSigningKey = key;
        bearerOptions.TokenValidationParameters.ValidateIssuerSigningKey = true;
    })
    .AddAuthorization();
```

**Comment by dj-nitehawk** [comment]
i believe that would break existing projects...
because currently, when it's a symmetric key, we expect the secret to be not base64 encoded.
this is only a concern because `user-jwts` stores the secret base64 encoded yes?
if so, wouldn't just decoding the secret yourself when reading it in work:
```cs
Encoding.UTF8.GetString(Convert.FromBase64String(builder.Configuration["..."]))
```

seems to be a [common issue](https://www.kallemarjokorpi.fi/blog/local-token-issuer/) with the `user-jwts` tool, and the above is the solution.

so, the current behavior will not be changed.

**Comment by queryquandary** [comment]
Thanks @dj-nitehawk
You're right - my proposed change could potentially break existing projects that use AddAuthenticationJwtBearer() with symmetric keys.  What I didn't know while writing my previous message is that Microsoft.AspNetCore.Authentication.JwtBearer.JwtSecurityTokenHandler v8.0.11 will go hunting for a signing key in configuration if the one specified in jwtSigningOptions.SigningKey doesn't work, and I happened to have the correct key set up in user secrets.  That's what lead me to suggest it worked when the api targets net8.0.  Microsoft.AspNetCore.Authentication.JwtBearer.JwtSecurityTokenHandler v9.0.0 doesn't seem to hit configuration, at least not the way 8.0.11 did as far as I can tell.

I also didn't know AddAuthenticationJwtBearer(signingOptions) was assuming symmetric keys would always be ASCII strings.  I made the incorrect assumption they would be base 64 encoded.  I typically use asymmetric keys and auth isn't something I spend much time on so I've learned something here.

Base64 string to ASCII bytes is a lossy conversion since ASCII only supports characters from 0-127 and anything outside that range gets substituted with 0x3f.  For that reason I don't think there's any way to use a base64 encoded signing key in AddAuthenticationJwtBearer(signingOptions) without manually replacing TokenValidationParameters.IssuerSigningKey with a new SymmetricSecurityKey().

There are lots of easy ways to work around this so like you said, leaving current behavior makes sense.
